### PR TITLE
ci: Run Datadog SCA in CI

### PR DESCRIPTION
**Pull request by harmonherring-pro**
### What does this PR do?
Adds a new Gitlab CI job that dogfoods the Datadog SCA product

### Motivation
@DataDog/software-integrity-and-trust partners with @DataDog/static-analysis to dogfood their SCA product and secure Datadog's supply chain.

### Reviewer's Checklist

- [ ] Changed code has unit tests for its functionality at or near 100% coverage.
- [ ] [System-Tests](https://github.com/DataDog/system-tests/) covering this feature have been added and enabled with the va.b.c-dev version tag.
- [ ] There is a benchmark for any new code, or changes to existing code.
- [ ] If this interacts with the agent in a new way, a system test has been added.
- [ ] Add an appropriate team label so this PR gets put in the right place for the release notes.
- [ ] Non-trivial go.mod changes, e.g. adding new modules, are reviewed by @DataDog/dd-trace-go-guild.

### CODE REVIEW

**Comment by Julio-Guerra** [comment]
Cool to see this initiative poping here ;-)

Is the docker image publicly accessible? If so, I suggest we do a github workflow instead, as this is our main CI tool.

**Comment by juli1** [comment]
> Cool to see this initiative poping here ;-)
> 
> Is the docker image publicly accessible? If so, I suggest we do a github workflow instead, as this is our main CI tool. (You could even provide a simple reusable github workflow or action to hide all these details).

It's not. If you want to use a GitHub action, you can use the [SCA GitHub action](https://github.com/DataDog/datadog-sca-github-action).

**Comment by Julio-Guerra** [comment]
> > Cool to see this initiative poping here ;-)
> > Is the docker image publicly accessible? If so, I suggest we do a github workflow instead, as this is our main CI tool. (You could even provide a simple reusable github workflow or action to hide all these details).
> 
> It's not. If you want to use a GitHub action, you can use the [SCA GitHub action](https://github.com/DataDog/datadog-sca-github-action).

So why not integrate this here? We are an open-source library, so the more open we are, the better ;-)
So far GitLab is used only for the benchmark platform which relies on gitlab runners maintained by the related backend team, everything else we have and rely on ourselves runs on GitHub Workflow.

Still open to starting your early integration this way.

**Comment by darccio** [comment]
I was planning to apply this at some point. Thanks for taking the lead, @harmonherring-pro! As @Julio-Guerra stated, I think too that we should go with the SCA GitHub Action.

If you don't mind, I can take care of this and convert the GitLab YAML to its GitHub equivalent in this same PR. WDYT?

**Comment by harmonherring-pro** [comment]
> I was planning to apply this at some point. Thanks for taking the lead, @harmonherring-pro! As @Julio-Guerra stated, I think too that we should go with the SCA GitHub Action.
> 
> If you don't mind, I can take care of this and convert the GitLab YAML to its GitHub equivalent in this same PR. WDYT?

Thanks for pinging me! I was wrapped up in some other work for a few days and didn't realize the activity this PR had 😅 I'll convert this PR and request your review on it when that's done. Thanks!

File: .gitlab/software_composition_analysis.yaml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  script:
+    # Disabling tracing to avoid leaking secrets.
+    # See https://www.gnu.org/software/bash/manual/bash.html#The-Set-Builtin:
+    # "Using ‘+’ rather than ‘-’ causes these options to be turned off"
+    - set +o xtrace
+    - export DD_API_KEY=$(aws ssm get-parameter --region us-east-1 --name "ci.dd-trace-go.datadog_api_key_org2" --with-decryption --query "Parameter.Value" --out text)
+    - export DD_APP_KEY=$(aws ssm get-parameter --region us-east-1 --name "ci.dd-trace-go.datadog_app_key_org2" --with-decryption --query "Parameter.Value" --out text)
+    - set -o xtrace
+    - osv-scanner --skip-git --recursive --experimental-only-packages --format=cyclonedx-1-4 --output=/tmp/sbom.json .
+    - datadog-ci sbom upload --service integrations-core --env ci /tmp/sbom.json
```
Comment (juliendoutre) [inline .gitlab/software_composition_analysis.yaml:19]: Could you set the service to `dd-trace-go` please?

**Review by juliendoutre — APPROVED** [review]
Looks good to me! Approving just in case, but feel free to go with the github actions integration if you want to.

**Pull request closed without merging**